src/components/Select/Select.js

```diff
@@ -0,0 +1,147 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import styled, { css, cx } from 'react-emotion';
+
+import { textMega } from '../../styles/style-helpers';
+import IconInputWrapper from '../IconInputWrapper';
+import DownTriangleIcon from './down-triangle.svg';
+
+const baseSelectStyles = ({ theme }) => css`
+  label: select;
+  appearance: none;
+  background-color: ${theme.colors.white};
+  border-width: 1px;
+  border-style: solid;
+  border-color: ${theme.colors.n300};
+  border-radius: ${theme.borderRadius.mega};
+  box-shadow: inset 0 1px 2px 0 rgba(102, 113, 123, 0.21);
+  color: ${theme.colors.n900};
+  padding: ${theme.spacings.byte} ${theme.spacings.tera} ${theme.spacings.byte}
+    ${theme.spacings.kilo};
+  position: relative;
+  ${textMega({ theme })};
+  z-index: 30;
+
+  &:focus,
+  &:hover,
+  &:active {
+    outline: none;
+  }
+
+  &:focus {
+    border: 1px solid ${theme.colors.b500};
+  }
+
+  &::after {
+  }
+
+  &:-moz-focusring {
+    color: transparent;
+    text-shadow: 0 0 0 #000;
+  }
+`;
+
+const disabledSelectStyles = ({ disabled }) =>
+  disabled &&
+  css`
+    label: select--disabled;
+    opacity: 0.4;
+    pointer-events: none;
+  `;
+
+const baseIconStyles = ({ theme }) => css`
+  label: select__icon;
+  fill: ${theme.colors.n700};
+  display: block;
+  z-index: 40;
+  pointer-events: none;
+`;
+
+const SelectElement = styled('select', { label: 'SelectElement' })(
+  baseSelectStyles,
+  disabledSelectStyles
+);
+const Icon = styled(DownTriangleIcon, { label: 'SelectIcon' })(baseIconStyles);
+
+/**
+ * A native select component.
+ */
+const Select = ({
+  options,
+  value,
+  placeholder,
+  selector,
+  disabled,
+  ...props
+}) => (
+  <IconInputWrapper
+    {...{ selector }}
+    iconPosition="right"
+    icon={({ className, disabledClassName }) => (
+      <Icon className={cx(className, { [disabledClassName]: disabled })} />
+    )}
+    input={({ className }) => (
+      <SelectElement {...{ ...props, className, disabled }}>
+        {!value && <option key={0}>{placeholder}</option>}
+        {options &&
+          options.map(option => (
+            <option key={option.value} {...option}>
+              {option.label}
+            </option>
+          ))}
+      </SelectElement>
+    )}
+  />
+);
+
+Select.propTypes = {
+  /**
+   * An ID rendered as data-selector attribute on the
+   * component. Used for tracking and e2e testing.
+   */
+  selector: PropTypes.string.isRequired,
```

Comment:
Something to discuss, I know it's not final - but I wouldn't want to assume that click-tracking and e2e testing should share a 'selector'. Often we may want new events to test different features, but perhaps can still not break our tests, for instance.

What about taking an object maybe? like `<Select selector={{ track: 'shipping-select@signup', test: 'signup-shipping-country' }}/>` which would end up with `<select data-track="" data-test="">`?

And then of course we have to decide naming conventions, but this would allow us to decouple data-* attributes for different uses.

Reply:
Hm good point. So far my assumption/thinking here was that
- This is really just a selector that we use instead of an `id` attribute.
- Handling of the event naming etc. should be taken care of in Mixpanel or whatever tracking tool we use.
- All that matters for testing is having a unique identifier.

Let's discuss in the 🍅 🆙 this afternoon.